corehq/apps/userreports/sql/data_source.py

```diff
@@ -87,7 +87,13 @@ def get_total_records(self):
             qc.append_column(c.view)
 
         session = connection_manager.get_scoped_session(self.engine_id)
-        return qc.count(session.connection(), self.filter_values)
+        try:
+            return qc.count(session.connection(), self.filter_values)
+        except Exception:
+            session.rollback()
+            raise
```

Comment:
Looks like [`session.remove()`](http://docs.sqlalchemy.org/en/latest/orm/contextual.html#sqlalchemy.orm.scoping.scoped_session.remove) calls `rollback()`, so this exception handler is not necessary.

Same below.

Reply:
I think it makes sense to use the session context here as well rather than duplicate the exception handling

Reply:
Did this the same way you did in your other PR